flasher.cpp

```diff
@@ -130,6 +132,10 @@ void Flasher::Init()
         socket_client_ = std::make_shared<socket::SocketClient>(servers_array);
     }
 
+    file_downloader_ = new file_downloader::FileDownloader();
```

Comment:
Memory leak.
file_downloader_ should be deleted at the end of the program. My suggestion is to use smart pointer here.